### PR TITLE
Add Nv/AMD sycl target build cmd

### PR DESCRIPTION
**Pull request by abhilash1910**
Cmake modification for Nv/AMD SYCL builds. 
@NeoZhangJianyu @ggerganov  @airMeng @AidanBeltonS @Alcpz

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -520,7 +520,7 @@ if (LLAMA_SYCL)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-narrowing")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl -L${MKLROOT}/lib")
-
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl-targets=nvptx64-nvidia-cuda,amdgcn-amd-amdhsa")
```
[the remark this answers is not in the document]
Comment (abhilash1910) [reply CMakeLists.txt:523]: Yes that makes sense.

**Comment by abhilash1910** [comment]
For SYCL runtime perf across nv and amd vendors this is based on sycl.cpp codebase. Features which will get added to the sycl code base will be tested as is on the other vendors . Priority wise we support Intel GPUs optimization and see performance of codebase on nv and amd. on sycl runtime.

**Comment by NeoZhangJianyu** [comment]
@abhilash1910 
1. Please make sure test is passed with NV & AMD GPU.
2. Update the README-sycl.md to guide how to install related software, build and run.
3. Add the GPU modes verified by you.
4. Run the CI by ci/run.sh on NV & AMD GPU.

**Comment by Alcpz** [comment]
@abhilash1910 @AidanBeltonS has been trying to run the SYCL version on Nvidia GPUs, but the tests are still not passing.
Another issue is that AMD builds require to manually specify the `--offload-arch` manually as there is currently no default parameter for that. We will be aiding with the review shortly.

**Comment by AidanBeltonS** [comment]
> I tried running it on Nvidia and AMD on Linux, but ran into some issues. It takes very long to compile and still only reports the Intel/CPU devices in the system, then dies.
> 
> ```
> Native API failed. Native API returns: -42 (PI_ERROR_INVALID_BINARY) -42 (PI_ERROR_INVALID_BINARY)Exception caught at file:/home/user/upstream-llama.cpp/ggml-sycl.cpp, line:12706
> ```
When you build for non spirv targets (i.e. NVidia and AMD) you must pass the device triple to the compiler i.e. `-fsycl-targets=nvptx64-nvidia-cuda` and `-fsycl-targets=amdgcn-amd-amdhsa -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx90a` for Nvidia and AMD respectively. Please note, you must replace `gfx90a` with your AMD GPU's architecture. Then you will no longer get an invalid binary error.

However, as pointed out before NVidia and AMD are not yet passing all tests so you should not expect it to run properly just yet.

**Comment by NeoZhangJianyu** [comment]
> > I tried running it on Nvidia and AMD on Linux, but ran into some issues. It takes very long to compile and still only reports the Intel/CPU devices in the system, then dies.
> > ```
> > Native API failed. Native API returns: -42 (PI_ERROR_INVALID_BINARY) -42 (PI_ERROR_INVALID_BINARY)Exception caught at file:/home/user/upstream-llama.cpp/ggml-sycl.cpp, line:12706
> > ```
> 
> When you build for non spirv targets (i.e. NVidia and AMD) you must pass the device triple to the compiler i.e. `-fsycl-targets=nvptx64-nvidia-cuda` and `-fsycl-targets=amdgcn-amd-amdhsa -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx90a` for Nvidia and AMD respectively. Please note, you must replace `gfx90a` with your AMD GPU's architecture. Then you will no longer get an invalid binary error.
> 
> However, as pointed out before NVidia and AMD are not yet passing all tests so you should not expect it to run properly just yet.

@abhilash1910 @AidanBeltonS 
Is it possible to update such compile setting info in the CMakeFile.txt or README-sycl.md?
so that reduce same issue.

I suggest adding a sub chapter "AOT" in chapter "build" in README-sycl.md.

**Comment by AidanBeltonS** [comment]
> @abhilash1910 @AidanBeltonS Is it possible to update such compile setting info in the CMakeFile.txt or README-sycl.md? so that reduce same issue.
> 
> I suggest adding a sub chapter "AOT" in chapter "build" in README-sycl.md.

Yes, I think we should update the CMake and README to properly support this. However, I do not propose making this change until we have the CUDA and HIP backends passing tests which is currently not the case.

**Comment by abhilash1910** [comment]
Addressed in #5738.Closing

**Pull request closed without merging**